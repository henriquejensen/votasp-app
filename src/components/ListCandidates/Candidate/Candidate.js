import React from 'react';
import './Candidate.scss';

const Candidate = ({
  name,
  number,
  politicalParty,
  picture = 'http://via.placeholder.com/150x150?text=Foto',
  status,
}) => (
    <div className="deputado-list-item">
      <div className="photo">
        <img src={picture} alt={`Foto do candidato ${name}`} />
      </div>
      <div className="info">
        <span className="candidate-name">{name}</span>
        <span className="candidate-number">{number}</span>
        {status === "Cadastro Finalizado"
          ? <span className="candidate-status bold green">{status}</span>
          : <span className="candidate-status bold red">{status}</span>
        }
        <div>
          <span className="partido-name">{politicalParty}</span>
        </div>
      </div>
    </div>
  )

export default Candidate;
