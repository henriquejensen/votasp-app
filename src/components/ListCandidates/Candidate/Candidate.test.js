import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Candidate from './Candidate';

describe('Candidate', () => {
  it('should match a snapshot', () => {
    const name = 'Candidate Name';
    const number = 1234;
    const politicalParty = 'PARTY';
    const picture = 'http://via.placeholder.com/150x150?text=TEST';
    const status = "Cadastro Finalizado";
    expect(
      toJson(
        shallow(( 
          <Candidate
            name={name}
            number={number}
            politicalParty={politicalParty}
            picture={picture}
            status={status}
          /> 
        ))
      )
    ).toMatchSnapshot();
  })

  it('should render correctly with no picture passed', () => {
    const name = 'Candidate Name';
    const number = 1234;
    const politicalParty = 'PARTY';
    const status = "Cadastro Finalizado";
    expect(
      toJson(
        shallow(( 
          <Candidate
            name={name}
            number={number}
            politicalParty={politicalParty}
            status={status}
          /> 
        ))
      )
    ).toMatchSnapshot();
  })

  it('should render correctly with no picture passed', () => {
    const name = 'Candidate Name';
    const number = 1234;
    const politicalParty = 'PARTY';
    const status = "Cadastro em andamento";
    expect(
      toJson(
        shallow(( 
          <Candidate
            name={name}
            number={number}
            politicalParty={politicalParty}
            status={status}
          /> 
        ))
      )
    ).toMatchSnapshot();
  })
})
