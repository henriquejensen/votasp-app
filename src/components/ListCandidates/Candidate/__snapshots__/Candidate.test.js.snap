// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidate should match a snapshot 1`] = `
<div
  className="deputado-list-item"
>
  <div
    className="photo"
  >
    <img
      alt="Foto do candidato Candidate Name"
      src="http://via.placeholder.com/150x150?text=TEST"
    />
  </div>
  <div
    className="info"
  >
    <span
      className="candidate-name"
    >
      Candidate Name
    </span>
    <span
      className="candidate-number"
    >
      1234
    </span>
    <span
      className="candidate-status bold green"
    >
      Cadastro Finalizado
    </span>
    <div>
      <span
        className="partido-name"
      >
        PARTY
      </span>
    </div>
  </div>
</div>
`;

exports[`Candidate should render correctly with no picture passed 1`] = `
<div
  className="deputado-list-item"
>
  <div
    className="photo"
  >
    <img
      alt="Foto do candidato Candidate Name"
      src="http://via.placeholder.com/150x150?text=Foto"
    />
  </div>
  <div
    className="info"
  >
    <span
      className="candidate-name"
    >
      Candidate Name
    </span>
    <span
      className="candidate-number"
    >
      1234
    </span>
    <span
      className="candidate-status bold green"
    >
      Cadastro Finalizado
    </span>
    <div>
      <span
        className="partido-name"
      >
        PARTY
      </span>
    </div>
  </div>
</div>
`;

exports[`Candidate should render correctly with no picture passed 2`] = `
<div
  className="deputado-list-item"
>
  <div
    className="photo"
  >
    <img
      alt="Foto do candidato Candidate Name"
      src="http://via.placeholder.com/150x150?text=Foto"
    />
  </div>
  <div
    className="info"
  >
    <span
      className="candidate-name"
    >
      Candidate Name
    </span>
    <span
      className="candidate-number"
    >
      1234
    </span>
    <span
      className="candidate-status bold red"
    >
      Cadastro em andamento
    </span>
    <div>
      <span
        className="partido-name"
      >
        PARTY
      </span>
    </div>
  </div>
</div>
`;
